utils/src/main/java/org/corfudb/utils/lock/LockClient.java

```diff
@@ -129,6 +140,24 @@ public void registerInterest(@NonNull String lockGroup, @NonNull String lockName
         monitorLocks();
     }
 
+    /**
+     * Enforce a transition to NoLeaseState and stop monitoring.
+     **/
+    public void suspendInterest() {
+        log.info("LockClient {} suspendInterest.", clientId);
+        stopMonitorLocks();
```

Comment:
stopMonitorLocks() only works on locks with expired lease.  Here we want to revoke lease on active locks.  So we need to generate an event "lease_revoked".. right?

Reply:
monitor thread monitors active locks too.